app/lib/utils/gpDataMapper.js

```diff
@@ -3,4 +3,58 @@ function getBookOnlineLink(gpData) {
     ? gpData.onlineServices.appointments.url : undefined;
 }
 
-module.exports = getBookOnlineLink;
+function getFilteredGps(gpData, searchStr) {
+  if (gpData.doctors) {
+    let filteredSearchStr = searchStr;
+    let filteredDocs = [];
+    if (searchStr.match(/^DOCTOR|Doctor|doctor|DR|Dr.|Dr|dr.|dr/)) {
```

Comment:
Could/should the match regex be done case insensitively? Would some of the duplicated `dr` be able to be removed?